samples/apps/copilot-chat-app/webapp/src/components/chat/ChatRoom.tsx

```diff
@@ -56,6 +58,41 @@ export const ChatRoom: React.FC = () => {
     const [isBotTyping, setIsBotTyping] = React.useState(false);
 
     const chat = useChat();
+        
+    const handleReceiveConversationMessageFE = (userAsk: UserAsk) => {
+        log("Received User Ask from backend");
+
+        const otherUsersAsk = {
+            timestamp: new Date().getTime(),
+            userId: 'otheruser',
+            userName: 'otheruser',
+            content: userAsk.input,
+            authorRole: AuthorRoles.User,
+        };
+        dispatch(updateConversation({ message: otherUsersAsk, chatId: selectedId }));
+    }
+    const handleReceiveChatSkillAskResultFE = (userAskResult: UserAskResult) => {
+        log("Received Chatbot Ask Result from backend");
+        
+        const messageResult = {
+            timestamp: new Date().getTime(),
+            userName: 'bot',
+            userId: 'bot',
+            content: userAskResult.value,
+            authorRole: AuthorRoles.Bot,
+        };    
+        dispatch(updateConversation({ message: messageResult, chatId: selectedId }));
+    }
+
+    React.useEffect(() => {
+        // This code will run once, when the component is mounted
+        console.log('SignalR setup called');
+        const { events } = SKMultiUserChatConnector();
```

Comment:
Moving conversation to this PR  @teresaqhoang / @TaoChenOSU / @adrianwyatt / @Vivihung 
The call to SKMultiUserChatConnector() should only happen once when the component is mounted. However there seems to be a bug somewhere thats making the component get mounted twice.